tunnel/intra/tcp.go

```diff
@@ -19,36 +19,81 @@ package intra
 import (
 	"io"
 	"net"
+	"time"
 
 	"github.com/eycorsican/go-tun2socks/common/log"
 	"github.com/eycorsican/go-tun2socks/core"
 )
 
 type tcpHandler struct {
-	fakedns net.Addr
-	truedns net.Addr
+	fakedns  net.Addr
+	truedns  net.Addr
+	listener TCPListener
+}
+
+// Usage summary for each TCP socket, reported when it is closed.
+type TCPSocketSummary struct {
+	Download   int64 // Total bytes downloaded.
```

Comment:
Consider appending the unit to the name (i.e. UploadBytes)

Reply:
Done